mod/groups/actions/groups/membership/join.php

```diff
@@ -44,35 +44,40 @@
 			register_error(elgg_echo("groups:cantjoin"));
 		}
 	} else {
-		add_entity_relationship($user->guid, 'membership_request', $group->guid);
```

Comment:
Instead use a return early strategy, which (in addition to other benefits) makes the diff cleaner to read. E.g.

```php
if (check_entity_relationship($user->guid, 'membership_request', $group->guid)) {
	register_error(elgg_echo("groups:requestexists"));
	forward(REFERER);
}
```

Also we enforce a [code style](http://learn.elgg.org/en/stable/contribute/code.html#php-guidelines) close to PSR-2. The Travis build will tell you if you get it wrong.